index.js

```diff
@@ -432,7 +465,10 @@
     'String#fantasy-land/concat': function(s) { return this + s; },
 
     //  String#empty :: String ~> String
-    'String#fantasy-land/empty': function() { return ''; }
+    'String#fantasy-land/empty': function() { return ''; },
+
+    //  Undefined#equals :: Undefined ~> Undefined -> Boolean
+    'Undefined#fantasy-land/equals': function(other) { return true; }
```

Comment:
`Z.equals` contains the guard `type(x) === type(y)`. If `x` and `y` are not members of the same type we don't dispatch to `fantasy-land/equals`. This means that each equality function is only responsible for comparing two members of its type. For example:

```haskell
Undefined#fantasy-land/equals :: Undefined ~> Undefined -> Boolean
```

There is only one member of the Undefined type, so checking the value is unnecessary. We could use `other === undefined` instead, but it would then be easy to assume that the method's domain is all values rather than all values of the Undefined type.